src/lib.rs

```diff
@@ -586,22 +593,54 @@ mod tests {
     fn parse_csi_max_params() {
         use crate::MAX_PARAMS;
 
-        static INPUT: &[u8] = b"\x1b[1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;1;p";
+        // This will build a list of repeating '1;'s
+        // The length is MAX_PARAMS - 1 because the last semicolon is interpreted
+        // as an implicit zero, making the total number of parameters MAX_PARAMS
+        let params = std::iter::repeat("1;").take(MAX_PARAMS - 1).collect::<std::string::String>();
```

Comment:
This line got pretty long, if we extract std::iter and std::string::String as imports it should be a bit easier to read.

I'd assume the full `String` path is necessary because of the `#![no_std]` by default, in which case it might make sense to import `use std::prelude::v1::*;`, since that automatically imports everything we get in a normal `std` environment.